src/VirtualClient/VirtualClient.Core/UnixDiskManager.cs

```diff
@@ -169,6 +169,45 @@ public override Task<IEnumerable<Disk>> GetDisksAsync(CancellationToken cancella
             });
         }
 
+        /// <summary>
+        /// Grabs the available data directory on the system.
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <param name="diskFilter">Disk filter to </param>
+        /// <returns></returns>
+        /// <exception cref="WorkloadException"></exception>
+        public override Task<IEnumerable<string>> GetDiskPathsAsync(string diskFilter, CancellationToken cancellationToken)
+        {
+            List<string> diskPaths = new List<string>();
+
+            return this.Logger.LogMessageAsync($"{nameof(UnixDiskManager)}.GetDiskPaths", EventContext.Persisted(), async () =>
+            {
+                IEnumerable<Disk> disks = await this.GetDisksAsync(cancellationToken).ConfigureAwait(false);
+                IEnumerable<Disk> disksToTest = DiskFilters.FilterDisks(disks, diskFilter, PlatformID.Unix).ToList();
+
+                if (disksToTest?.Any() != true)
+                {
+                    throw new WorkloadException(
+                        "Expected disks to test not found. Given the parameters defined for the profile action/step or those passed " +
+                        "in on the command line, the requisite disks do not exist on the system or could not be identified based on the properties " +
+                        "of the existing disks.",
+                        ErrorReason.DependencyNotFound);
+                }
+
+                foreach (Disk disk in disksToTest)
```

Comment:
If I'm understanding correctly, you just need the extension methods on Disks -> Disk.GetPreferredAccessPath? You don't need to put that into DiskManager if that's the case

Reply:
GetPreferredAccessPath is leveraged to get the access paths of all the disks attached / filtered by DiskFilter. Since this method appears in >2 workloads & dependencies, thought it would be appropriate as a method through the disk manager

Reply:
If that's the case you should call the GetDisks function above to avoid the duplicated filter code